### DISCUSSION TITLE
Welcome to gdext-nim Discussions!

### DISCUSSION THREAD

**Discussion by panno8M** [godot-nim/godot#128] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by xix-xeaon** [comment]
I'm a teacher of programming. At my school we normally teach Python, which is good because it's beginner friendly. However, that also makes it more difficult to teach some parts of the curriculum dealing with the lower levels. An other language, like C, needs to be included as well but this confuses the students.

Nim could be an alternative which is both reasonably beginner friendly, and covers more parts of the curriculum. If we could also include some game development then it would certainly motivate the students and hopefully contribute to successful learning. I've been trying out various random things with this extension for this reason and I'm optimistic. Of course, I doubt the school will switch from Python to nim in general, but at least I can include it in my own classroom instead of C.

I think it would bring in a lot more people to this community if there were a few simple but good games made using it. Unfortunately I'm mostly busy with teaching and other projects.

**Reply by panno8M** [reply]
Certainly Nim is the best choice as an intermediate between Python and C. It is also good that the syntax learned in Python is hardly wasted.

When I get time, I will add games like tetris or brick breaker to the demo.

**Reply by insomniacUNDERSCORElemon** [reply]
I've never really been that interested in Python due to speed/power, though the newer changes (JIT, disabling GIL) probably could change that. 

Though that seems to only complicate Godot support, so Nim seems like a better option here.

**Reply by insomniacUNDERSCORElemon** [reply]
> I think it would bring in a lot more people to this community if there were a few simple but good games made using it

> When I get time, I will add games like tetris or brick breaker to the demo.

I just saw news of a PR adding Jolt physics being merged. If it makes it into 4.4, it seems like this could be useful as a foundation for a 3D demo. Be that any type of game played directly with physics-input\*, part of game-play like HL2, or just background/juice-factor (destructibles/sandbox stuff)

\*= pinball, pachinko, marble maze/race/platforming, pool, golf?, racing

EDIT: And maybe more (water, conveyor belts, softbodies), also 4.4 seems to have better CSG support

**Comment by panno8M** [comment]
Hi there! I am a Japanese university student and I am the leader of the project.
In university, I majored in Programming and 3D Graphics. This project was originally started as a university project. It was then open-sourced and generalized to become gdext-nim.

This project is currently settled, but I will try to respond to your bug reports and feature requests as much as possible.

Cheers!

**Reply by zhmt** [reply]
My questions are often answered during the day, so I suspect the author's time zone is not far from mine, and it turns out to be true, LOL. I am from China.

**Comment by insomniacUNDERSCORElemon** [comment]
I'm a long-time lurker with Godot and Nim, still not really started. No real end goal other than something to do. The issue for me now is mostly getting motivation for some low-scope project and how to make it actually fun.

I have been tinkering with visuals. I made the stone head below to test out working with general materials that use a texture but didn't really like the 2 versions I made for multiple reasons (and did not get the desired effect). I started leaning more towards an unshaded material, but I did add some (mostly) smooth shading via the vertex colors.

![A low-poly, vertex color Moai-esque stone head with glass eyes. The scene has reduced color for lower data usage, resulting in dithering](https://github.com/user-attachments/assets/c72e9263-0845-43de-8d7e-89b0301f8d30)

There are other puzzle pieces I am not sure of, like gridmaps seem alright but I'm thinking other tech might be better for other types content.

EDIT: There's also Blender tech (related to VC workflow) that I have not tried yet. I'm especially interested in if the newer stuff works, like using greasepencil for generating an animated mesh. In the past I've tried using multiple VC maps arranged with a Blender visual shader (one map for glow) and it didn't seem to export properly.

**Comment by zhmt** [comment]
I am a game developer from china. I am developing a dungeon action RPG, maybe with network support later.

I am very satisfied with the Nim language; it is quite fast and easy to compile alongside C. However, there is one aspect that frustrates me: circular dependencies. I don't know how you are currently handling them, but my current approach is to use abstract interfaces. However, this is not conducive to writing code quickly, as it adds a lot of extra workload and makes it difficult to deal with some code that has severe circular dependencies.

I have seen the official plans for Nim, and unfortunately, the priority for resolving circular dependencies is quite low.
